src/OpenSearch.Net.Auth.AwsSigV4/AwsSigV4HttpClientHandler.cs

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+#if DOTNETCORE
+
+namespace OpenSearch.Net.Auth.AwsSigV4
+{
+	using System;
+	using System.Net.Http;
+	using System.Threading;
+	using System.Threading.Tasks;
+	using Amazon;
+	using Amazon.Runtime;
+
+	internal class AwsSigV4HttpClientHandler : DelegatingHandler
+	{
+		private readonly AWSCredentials _credentials;
+		private readonly RegionEndpoint _region;
+
+		public AwsSigV4HttpClientHandler(AWSCredentials credentials, RegionEndpoint region, HttpMessageHandler innerHandler)
```

Comment:
I'd like default credentials but a configurable region.

Reply:
This particular class is a fully internal implementation detail where the credentials and region are already known.